server/app/controllers/MaintenanceController.scala

```diff
@@ -1,15 +1,74 @@
 package controllers
 
+import com.alexitc.playsonify.models.pagination.Limit
+import com.xsn.explorer.data.async.{BlockFutureDataHandler, TransactionFutureDataHandler}
+import com.xsn.explorer.gcs.{GolombCodedSet, GolombEncoding}
+import com.alexitc.playsonify.core.FutureOr.Implicits.FutureOps
+import com.xsn.explorer.data.anorm.AnormPostgresDataHandler
+import com.xsn.explorer.data.anorm.dao.BlockFilterPostgresDAO
+import com.xsn.explorer.executors.DatabaseExecutionContext
+import com.xsn.explorer.models.values.{Blockhash, Height}
 import controllers.common.{MyJsonController, MyJsonControllerComponents}
 import javax.inject.Inject
 import org.scalactic.Good
+import play.api.db.Database
 import play.api.libs.json.JsObject
 
 import scala.concurrent.Future
 
-class MaintenanceController @Inject()(components: MyJsonControllerComponents) extends MyJsonController(components) {
+class MaintenanceController @Inject()(
+    blockDataHandler: BlockFutureDataHandler,
+    transactionDataHandler: TransactionFutureDataHandler,
+    db: DBMigration,
+    components: MyJsonControllerComponents
+) extends MyJsonController(components) {
 
   def run(query: String) = public { _ =>
+    val startingHeight = query.toInt
+
+    logger.info(s"Migrating...")
+
+    (startingHeight to 0 by -1).foreach(height => {
+      logProgress(startingHeight, height)
+
+      for {
+        block <- blockDataHandler.getBy(Height(height)).toFutureOr
+        transactions <- transactionDataHandler.getTransactionsWithIOBy(block.hash, Limit(100000), None).toFutureOr
+        blockWithTransactions = block.withTransactions(transactions)
+        filter = GolombEncoding.encode(blockWithTransactions)
+        _ <- Future.successful(Good(db.updateBlockFilter(block.hash, filter.get))).toFutureOr
+      } yield ()
+    })
+
+    logger.info(s"Migration completed")
+
     Future.successful(Good(JsObject.empty))
   }
+
+  def logProgress(startingHeight: Int, currentHeight: Int) = {
+    val blockToMigrate: Int = startingHeight + 1
+    val migratedBlocks: Int = startingHeight - currentHeight
+    val percentage: Int = 100 * migratedBlocks / blockToMigrate
+    val previousPercentage: Int = 100 * (migratedBlocks - 1) / blockToMigrate
+
+    if (percentage != 0 && percentage != previousPercentage) {
+      logger.info(s"${percentage}% done")
```

Comment:
Percentage for what? the idea of the progress is to be able to restart the job in case if fails or the app is restarted.